assets/js/blocks/product-query/inspector-controls/product-selector.tsx

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import { getProducts } from '@woocommerce/editor-components/utils';
+import { ProductResponseItem } from '@woocommerce/types';
+import { objectOmit } from '@woocommerce/utils';
+import { useState, useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import {
+	FormTokenField,
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalToolsPanelItem as ToolsPanelItem,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { ProductQueryBlock } from '../types';
+import { setQueryAttribute } from '../utils';
+
+export const ProductSelector = ( props: ProductQueryBlock ) => {
+	const { query } = props.attributes;
+
+	const [ productsList, setProductsList ] = useState< ProductResponseItem[] >(
+		[]
+	);
+
+	useEffect( () => {
+		getProducts( { selected: [] } ).then( ( results ) => {
+			setProductsList( results as ProductResponseItem[] );
+		} );
+	}, [] );
```

Comment:
I don't think it's needed here but just wanted to share my thoughts. Maybe we can create a custom hook to fetch and manage the productsList state to separate concerns and make the component more modular:
```jsx
const useProductsList = (): ProductResponseItem[] => {
  const [productsList, setProductsList] = useState<ProductResponseItem[]>([]);

  useEffect(() => {
    getProducts({ selected: [] }).then((results) => {
      setProductsList(results as ProductResponseItem[]);
    });
  }, []);

  return productsList;
};
```

And use it in the component like this:
```jsx
const productsList = useProductsList();
```